packages/xchain-ethereum/src/client.ts

```diff
@@ -576,67 +475,109 @@ export default class Client implements XChainClient, EthereumClient {
     if (!address) {
       return Promise.reject(new Error('address must be provided'))
     }
-
     const contract = new ethers.Contract(address, abi, this.provider).connect(this.getWallet())
     return contract[func](...params)
   }
 
   /**
-   * Send a transaction to the vault.
+   * Check allowance.
    *
-   * @param {VaultTxOpts} params The Vault transaction options.
-   * @returns {TransactionResponse}  The vault transaction result.
+   * @param {Address} spender The spender address.
+   * @param {Address} sender The sender address.
+   * @param {BaseAmount} amount The amount of token.
+   * @returns {boolean} `true` or `false`.
    */
-  vaultTx = async ({ address, amount, memo, overrides }: VaultTxOpts): Promise<TransactionResponse> => {
+  private isApproved = async (spender: Address, sender: Address, amount: BaseAmount): Promise<boolean> => {
     try {
       const txAmount = parseEther(baseToAsset(amount).amount().toFormat())
```

Comment:
[question] Is it safe to use [Bignumber's `toFormat()`](https://mikemcl.github.io/bignumber.js/#toFor) here and at other places? Does [Bignumber's `toString()`](https://mikemcl.github.io/bignumber.js/#toS) does not work?

Reply:
`toString()` will parse BigNumber to `1e-18` which is invalid decimal value.